linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -627,19 +635,101 @@ mod tests {
         let model = DecisionTree::params(2).max_depth(Some(2)).fit(&dataset);
 
         assert_eq!(&model.features(), &[8]);
+        assert_eq!(
+            &model.feature_importance(),
+            &[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]
+        );
     }
 
     #[test]
     /// Small perfectly separable dataset test
     ///
     /// This dataset of three elements is perfectly using the second feature.
     fn perfectly_separable_small() {
-        let data = array![[1., 2., 3.5], [1., 2., 3.5], [1., 3., 3.5]];
+        let data = array![[1., 2., 3.], [1., 2., 3.5], [1., 3., 4.]];
```

Comment:
This should be separable on the second or third feature. if we make third feature 3, 4, 3.5 then I think it's just the the second feature that should be used. I think I did the same thing by accident when tweaking the numbers.

Reply:
it should only use the third feature in this case. Update in the latest commit to [3,4,3.5]